### ISSUE TITLE
Inline SVG rendering changed

### ISSUE THREAD

**Issue by johanneskrtek** [electron/electron#8670]
* Electron version: 1.6.0
* Operating system: macOS 10.12.3

### Expected behavior

Inline SVGs should be rendered as it is in previous versions of Electron.

### Actual behavior

Inline SVGs seem to have a `stroke-width` of about `5px` and `shapeRendering: OptimizeSpeed`.

![1 6 0](https://cloud.githubusercontent.com/assets/419747/22887743/7b20cd54-f203-11e6-9796-672e5d319f68.png)

![1 4 15](https://cloud.githubusercontent.com/assets/419747/22887736/777bf69c-f203-11e6-99f4-0e67bb2fc23c.png)

### How to reproduce

Start Electron or an Electron app with inline SVGs in macOS 10.12.3.

P.S. The Developer Tools do not work at all since version 1.5.0 in my app (do not show any styles), in Electron 1.6.0 itself the selection tool (![bildschirmfoto 2017-02-13 um 15 59 41](https://cloud.githubusercontent.com/assets/419747/22888318/8c1657b2-f205-11e6-8c14-c14f02145274.png)) does not work.

**Comment by groundwater** [comment]
Hi @johanneskrtek - I booted Electron as above, with the exact same version and OS version. Everything looks fine on my end. The SVG looks normal, and I can open devtools.

If you have more information about how you're running Electron (e.g. non-standard arguments or settings) or have unique system settings please let us know.

**Comment by johanneskrtek** [comment]
Hi,

that’s how I can reproduce the issue(s):

```
$ cd tmp
$ node -v
v7.4.0
$ npm install electron@1.6.0
$ $(npm bin)/electron
```

![electron](https://cloud.githubusercontent.com/assets/419747/22925958/f027bc32-f2aa-11e6-908b-e945bad1e956.png)

I can open the dev tools and they look normal in 1.6.0. But when I click the selection tool I get the following error in the command line:

```
[8536:0214/114134.095263:ERROR:CONSOLE(3203)] "Protocol Error: the message without method: {"error":{"code":-32700,"message":"Message must be a valid JSON"}}", source: chrome-devtools://devtools/bundled/inspector.js (3203)
```

**Comment by MarshallOfSound** [comment]
@johanneskrtek Can you provide you current display resolution and your current DPI scaling settings

**Comment by johanneskrtek** [comment]
Display setups:
* External display, 1920 x 1200 (scale factor 1)
* Internal retina display, 2880 x 1800, scaled to 1440 x 900 (scale factor 2)

But it also occurs in all other available scaled resolutions on the retina display.

**Comment by groundwater** [comment]
Hey @johanneskrtek I would like to bisect where this error might be coming from. Can you reproduce this on another machine besides yours? Please use the same external monitor if possible.

**Comment by johanneskrtek** [comment]
Update: the SVG rendering issue (and the dev tools selection tool issue) only occurs when I start the app with `npm start` or Electron with `$(npm bin)/electron`. It does not occur in the dev build of the app I am working on.

`process.versions` shows identical versions for development mode and dev build.

I will try to reproduce the issue on another machine and keep you up to date.

**Comment by johanneskrtek** [comment]
Hi @groundwater, just tested on my old machine (MBP 2012) with the same external display.

I cannot reproduce the SVG issue there, neither on the internal nor the external display. It works without any problems both in Electron and in my app.

But I can reproduce the Developer Tools selection tool issue on both machines, shall I open a separate issue for that?

**Comment by groundwater** [comment]
Hi @johanneskrtek, I think we should separate the issues.

We're happy to tackle this one if we can figure out what's going on. If you're making progress on debugging it let's leave it open, otherwise I think we should close it until it's happens to someone else.

Thanks for your debugging help!

**Comment by lele85** [comment]
I have the same issue here, SVG rendering is broken and developer tools are not working correctly.

**Comment by johanneskrtek** [comment]
I encountered broken SVG rendering in Chrome 57.0.2987.98. I could not reproduce it so far, but this might be a hint that the SVG rendering issue has nothing to do with Electron itself.

<img width="176" alt="stackoverflow" src="https://cloud.githubusercontent.com/assets/419747/23943854/21dc2e2a-0971-11e7-98d6-f825dfb191e4.png">

**Comment by yarastqt** [comment]
Have similar problem, rendering svg is broken, and devtools not working correctly have error message in terminal:
```shell
[4278:0315/192233.421861:ERROR:CONSOLE(3203)] "Protocol Error: the message without method: {"error":{"code":-32700,"message":"Message must be a valid JSON"}}", source: chrome-devtools://devtools/bundled/inspector.js (3203)
```
MBP (2014) Mac OS 10.11.5

**Comment by Reggionick** [comment]
Same issues with MacBook Pro (Retina, 15-inch, Late 2013) and external monitor @ 1920 x 1200
DevTools not working and SVG rendering broken
![](https://www.dropbox.com/s/qtcq08jz2g88dpg/Screenshot%202017-03-16%2012.00.12.png?dl=1)

**Comment by mlajtos** [comment]
Just confirming the SVG and DevTools issue... But the rendering is sometimes okay and only after a while it goes bonkers.

![screenshot 2017-03-23 20 28 27](https://cloud.githubusercontent.com/assets/1047512/24266682/fb3fd0b2-1007-11e7-9518-7cfd7e5a1a2e.png)

**Comment by julienma** [comment]
I confirm I have the same SVG rendering issue:
- Electron version: 1.6.1
- Operating system: macOS 10.12.3
- app using react + webpack + https://github.com/gaearon/react-hot-loader with https://github.com/chentsulin/electron-react-boilerplate
- MBPr 2013

**However, SVG is OK in the production .app** built with `npm run package`, as already said in https://github.com/electron/electron/issues/8670#issuecomment-281170700.

**Comment by pixtron** [comment]
I can confirm the svg rendering issue and the chrome devtools errors.

The rendering bug only occurs when starting the app with `npm start` or when the app has been packaged with `$(npm bin)/build --dir`. 
When the app has been packaged with `$(npm bin)/build --mac dmg` and the app is started from the resulting dmg the icons are displayed correctly most of the time.


- Electron version 1.6.2 & 1.6.4
- OSX 10.11.6
- mbp mid 2012

**Comment by johanneskrtek** [comment]
Update: I do not run into the issue when I use English as macOS system language. @lele85, @yarastqt, @Reggionick, @mlajtos, @julienma, @pixtron – put your hand up, who of you does **not** use English as system language?

**Comment by mlajtos** [comment]
@johanneskrtek Guilty.

**Comment by pixtron** [comment]
@johanneskrtek i can confirm, that i don't run into the issue if i set osx system language to english. usually i'm using german.

**Comment by julienma** [comment]
@johanneskrtek yep, using French.

**Comment by johanneskrtek** [comment]
Thanks for the feedback.

Looks like this is the Chromium bug behind the SVG issue: https://bugs.chromium.org/p/chromium/issues/detail?id=698609&q=svg&colspec=ID%20Pri%20M%20Stars%20ReleaseBlock%20Component%20Status%20Owner%20Summary%20OS%20Modified

It can be worked around in Electron by disabling the hardware acceleration with `app.disableHardwareAcceleration()`.

**Comment by julienma** [comment]
Thanks for the tip. So I guess we can use `app.disableHardwareAcceleration()` in development mode only, as this issue does not appear when packaging the app for production.

Also might be worth reporting this thread in the chromium issue, to help these guys track it down?

**Comment by johanneskrtek** [comment]
Already done.

**Comment by kevinsawicki** [comment]
I believe this is similar to #9225 and should be resolved by https://github.com/electron/libchromiumcontent/pull/299 which was included in the 1.7.2 release which is now in beta, closing this out.
